src/Xamarin.Android.Build.Tasks/Tasks/CheckForInvalidResourceFileNames.cs

```diff
@@ -17,13 +17,22 @@ public class CheckForInvalidResourceFileNames : Task {
 		public ITaskItem[] Resources { get; set; }
 
 		Regex fileNameCheck = new Regex ("[^a-zA-Z0-9_.]+", RegexOptions.Compiled);
+		Regex fileNameWithHyphenCheck = new Regex ("[^a-zA-Z0-9_.-]+", RegexOptions.Compiled);
 
 		public override bool Execute ()
 		{
 			foreach (var resource in Resources) {
-				var match = fileNameCheck.Match (Path.GetFileName (resource.ItemSpec));
-				if (match.Success) {
-					Log.LogCodedError ("APT0000", resource.ItemSpec, 0, "Invalid file name: It must contain only [a-zA-Z0-9_.].");
+				var fileName = Path.GetFileName (resource.ItemSpec);
+				if (string.Compare (Directory.GetParent (resource.ItemSpec).Name, "values", StringComparison.OrdinalIgnoreCase) == 0) {
```

Comment:
...then I get *here*, and this `string.Compare()` call *won't* catch e.g. `values-es` as a resource directory name, meaning we'd hit the fallback path, meaning `Resources\values-es\strings-1.xml` is now an invalid name.

Is that desired/intended? If so, *why*?